File: todo_app/backend/src/utils/token.ts
import jwt from "jsonwebtoken";

const SECRET = process.env.JWT_SECRET || "todo-secret-key";

export const createToken = (userId: string): string => {
  return jwt.sign({ userId }, SECRET, { expiresIn: "7d" });
};

export const verifyToken = (token: string): { userId: string } | null => {
  try {
    return jwt.verify(token, SECRET) as { userId: string };
  } catch {
    return null;
  }
};

File: freelance_market_place/backend/src/utils/token.py
import jwt
import os
from src.db.schema import UserRole

JWT_SECRET = os.getenv("JWT_SECRET")

def create_token(user_id : int, email : str, role : UserRole):
    token = jwt.encode({
        "user_id" : user_id,
        "email" : email,
        "role" : role
    }, JWT_SECRET, algorithm="HS256")
    return token

def verify_token(token: str):
    check = jwt.decode(token, JWT_SECRET, ["HS256"])
    return check



